### ISSUE TITLE
Adding SetMinMaxZoom() API to PCL

### ISSUE THREAD

**Issue by idan-rubin** [amay077/Xamarin.Forms.GoogleMaps#315]
First of all I wanted to extend a huge THANK YOU for building this amazing package! It greatly reduces our custom native code and simplifies app.
----
Since both iOS and Android support min and max zoom for the map, can you please add it to the PCL?

which platform do you want?

- [x] Android
- [x] iOS

**Comment by amay077** [comment]
Yes we can, I found set min/max zoom-level in Android and iOS Google Maps API.

* [Android] GoogleMap.setMaxZoomPreference - https://developers.google.com/android/reference/com/google/android/gms/maps/GoogleMap
* [iOS] GMSMapView.setMinZoom - https://developers.google.com/maps/documentation/ios-sdk/reference/interface_g_m_s_map_view

**Comment by amay077** [comment]
I suggest adding listed APIs in Xamarin.Forms.GoogleMaps.

* ``Map.SetMinMaxZoomLevel(float minZoomLevel, float maxZoomLevel)`` method
* ``Map.MinZoomLevel`` property (getter only)
* ``Map.MaxZoomLevel`` property (getter only)

Why ``Map.MinZoom`` doesn't have the setter?
Because separated setter and getter to each methods in GoogleMaps Android API(``setMaxZoomPreference`` and ``getMinZoomLevel``).

But this implementations are very easy.
Is there somebody who can try?

**Comment by Shivaprabhu049** [comment]
@amay077 Have you implemented this Zoom Level setter properties.

**Comment by AbhishekSinghThakur** [comment]
Any one has implemented this I need to add this in my project

**Comment by AbhishekSinghThakur** [comment]
I have fixed this using customising maps CameraChanged method
`googleMap.CameraChanged += async (object sender, CameraChangedEventArgs e) => await GoogleMap_CameraChanged(sender,e) ;`

`async Task GoogleMap_CameraChanged(object sender, CameraChangedEventArgs e)
        {
            if(e.Position.Zoom>maxZoom){
                CameraPosition cameraPosition = new CameraPosition(e.Position.Target, maxZoom);
                CameraUpdate cameraUpdate = CameraUpdateFactory.NewCameraPosition(cameraPosition);
                googleMap.InitialCameraUpdate = cameraUpdate;
                await googleMap.AnimateCamera(cameraUpdate);
            }
            if (e.Position.Zoom <minZoom )
            {
                CameraPosition cameraPosition = new CameraPosition(e.Position.Target, minZoom);
                CameraUpdate cameraUpdate = CameraUpdateFactory.NewCameraPosition(cameraPosition);
                googleMap.InitialCameraUpdate = cameraUpdate;
                await googleMap.AnimateCamera(cameraUpdate);
            }
        }`

**Comment by galadril** [comment]
This feature would be really handy to have in this library

**Comment by valentasm1** [comment]
@AbhishekSinghThakur solution trigers infinitve loop. IMO AnimateCamera trigers CameraIdled which trigers Animate and so on. Such things eats batery life.

**Comment by valentasm1** [comment]
Come up with check wich prevents loop.
Of course 0.18 is very high but in my case it is ok. It should be as low as 0.02 to prevent loop
```
 var cameraIdlePosition = e.Position?.Target;
            if (cameraIdlePosition == null) return;
            if (_prevCameraPosition != null)
            {
                var distance = Location.CalculateDistance(_prevCameraPosition.Value.Latitude,
                    _prevCameraPosition.Value.Longitude, cameraIdlePosition.Value.Latitude,
                    cameraIdlePosition.Value.Longitude, DistanceUnits.Kilometers);
                if (distance < 0.18)
                {
                    _prevCameraPosition = e.Position?.Target;
                    return;
                }
            }

            _prevCameraPosition = e.Position?.Target;
```
